src/app/routes/cpsAsset/getInitialData/cpsRecomendations/index.js

```diff
@@ -0,0 +1,76 @@
+import path from 'ramda/src/path';
+import deepClone from 'ramda/src/clone';
+import splitAt from 'ramda/src/splitAt';
+import { optimoSubheadline } from '../convertToOptimoBlocks/utils/helpers';
+
+const isCpsParagraph = (block) => {
+  /* In CPS the following blocks are considered paragraphs
+    Image, Audio, Video, Paragraph
+  */
+  const mediaTypes = ['image', 'audio', 'video'];
+  if (mediaTypes.includes(block.type)) return true;
+
+  if (block.type === 'text') {
+    const paragraphBlocks = path(['model', 'blocks'], block).filter(
+      (b) => b.type === 'paragraph',
+    );
+    return !!paragraphBlocks.length;
+  }
+  return false;
+};
+
+const getNthCpsParagraphIndex = (blocks, n) => {
```

Comment:
I like this approach, I think it would be good to move `getNthCpsParagraphIndex` and `isCpsParagraph` to a new helper method, the logic for these methods can then be tested in isolation and also made available to other pre-process rules.